prov/verbs/src/fi_verbs.c

```diff
@@ -907,6 +936,16 @@ fi_ibv_create_ep(const char *node, const char *service,
 	if (ret)
 		return (errno == ENODEV) ? -FI_ENODATA : -errno;
 
+	/* Ignore ib_rai added by IBACM to prevent wrong ib_connect_hdr
+	 * from being sent in connect request. */
+	/* TODO Choose ib_rai if we came here from fi_endpoint */
+	if (_rai->ai_next) {
```

Comment:
It would be better to use a loop here and skip while _rai points to AF_IB, rather than assuming that ai_next means that we are pointing to AF_IB and the next one is not.